lisa/tools/hyperv.py

```diff
@@ -381,30 +399,59 @@ def create_virtual_disk(self, name: str, pool_name: str, columns: int = 2) -> No
             force_run=True,
         )
 
-    def _check_exists(self) -> bool:
-        try:
-            self.node.tools[PowerShell].run_cmdlet(
-                "Get-VM",
-                force_run=True,
-            )
-            self._log.debug("Hyper-V is installed")
-            return True
-        except LisaException as e:
-            self._log.debug(f"Hyper-V not installed: {e}")
-            return False
+    def configure_dhcp(self, dhcp_scope_name: str = "DHCPInternalNAT") -> None:
+        powershell = self.node.tools[PowerShell]
+        service: Service = self.node.tools[Service]
+
+        # Install DHCP server
+        self.node.tools[WindowsFeatureManagement].install_feature("DHCP")
```

Comment:
When is it needed? Hyper-V can assign IP addresses to VMs, it doens't need the DHCP server. When using the DHCP server, it's need to be careful, because it may conflict with lab DHCP.

Reply:
In Lab, nested VMs will get IPs from LAN DHCP using external Hyper-V Switch.
In Azure Windows Hyper-V hosts, We need to setup our own DHCP server with Internal Switch.

This will be used only in "HyperVPreparationTransformer" not in common Hyper-V environment setup.

Reply:
In Lab, nested VMs will get IPs from LAN DHCP using external Hyper-V Switch.
In Azure Windows Hyper-V hosts, We need to setup our own DHCP server with Internal Switch.

This will be used only in "HyperVPreparationTransformer" not in common Hyper-V environment setup.

Reply:
> In Azure Windows Hyper-V hosts, We need to setup our own DHCP server with Internal Switch.

Can you include a reference of official public document on how it works in the code comment? I need to take a look, and the code needs to be documented by comments also. My understanding is that the *internal* network is managed by HyperV, including IP address assignment. And the *external* switch doesn't work, if no extra IP address is assigned in the subnet on Azure.

Reply:
I added the reference doc to the PR description. Repasting it here as well.

https://techcommunity.microsoft.com/blog/itopstalkblog/how-to-setup-nested-virtualization-for-azure-vmvhd/1115338

Reply:
> " My understanding is that the internal network is managed by HyperV, including IP address assignment." - This may be true on Windows desktop versions but I found DHCP config needed on Win server 2025.

We have Windows Servers in lab, although they are not 2025, but I don't think it would be a regression to require DHCP. Did you try on Azure without DHCP server installed? The only benefit of the DHCP server is to specify the IP range, but it's not in our current test scope.


Add the link in the code comments instead of PR description. Please read the LISA documents about where to write documents. PR description is used to describe the change history, and the code comment is used to explain the current behavior.

Reply:
"Did you try on Azure without DHCP server installed?" I tried. L2 VMs are not getting IPs until I enable DHCP as mentioned the doc.

I added details about where this method should be used and shouldn't with a link to doc.